api4/channel.go

```diff
@@ -1112,24 +1112,27 @@ func addChannelMember(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Check join permission if adding yourself, otherwise check manage permission
-	if channel.Type == model.CHANNEL_OPEN {
-		if member.UserId == c.App.Session.UserId {
-			if !c.App.SessionHasPermissionToChannel(c.App.Session, channel.Id, model.PERMISSION_JOIN_PUBLIC_CHANNELS) {
-				c.SetPermissionError(model.PERMISSION_JOIN_PUBLIC_CHANNELS)
-				return
-			}
-		} else {
-			if !c.App.SessionHasPermissionToChannel(c.App.Session, channel.Id, model.PERMISSION_MANAGE_PUBLIC_CHANNEL_MEMBERS) {
-				c.SetPermissionError(model.PERMISSION_MANAGE_PUBLIC_CHANNEL_MEMBERS)
-				return
+	// Bypass permissions check if you are already joined to the channel
+	if _, err = c.App.GetChannelMember(member.ChannelId, member.UserId); err != nil {
```

Comment:
This /smells/ like a security issue: if `GetChannelMember` fails for any reason, we enter this block. Should we distinguish some `ErrorNotFound` from other errors?

Reply:
Good point

Reply:
Fixed

Reply:
Fixed